### PR TITLE
Show emoji in HTML treeview

### PR DESCRIPTION
**Pull request by albert-github**
Show emoji in the HTML treeview as emoji and not as  text

Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/14013838/example.tar.gz)

### CODE REVIEW

File: src/textdocvisitor.cpp [hunk 1]
```diff
@@ -39,10 +39,29 @@ void TextDocVisitor::operator()(const DocSymbol &s)
 
 void TextDocVisitor::operator()(const DocEmoji &s)
 {
-  const char *res = EmojiEntityMapper::instance().name(s.index());
+  // the TextDocVisitor is only invokated for the JS part of the HTML output
```
Comment (doxygen) [inline src/textdocvisitor.cpp:42]: typo: invokated->invoked

File: src/util.cpp [hunk 2]
```diff
@@ -5635,7 +5637,8 @@ QCString parseCommentAsText(const Definition *scope,const MemberDef *md,
       if (result.at(i)==',' ||
           result.at(i)=='.' ||
           result.at(i)=='!' ||
-          result.at(i)=='?')
+          result.at(i)=='?' ||
+          result.at(i)=='}')    // good for UTF-16 characters and } otherwise also hood point to stop string
```
Comment (doxygen) [inline src/util.cpp:5641]: typo: hood -> good

**Comment by doxygen** [comment]
@albert-github I think there is an issue with emoticons in section titles when using latex output as shown by your example (with and without this change), I get:

```
! Argument of \@sect has an extra }.
<inserted text> 
                \par 
l.3 ...tion \+:+1\+: \doxygenemoji{\+:+1\+:}{+1}}}
                                                  \label{pg_all_autotoc_md1}

```

**Comment by albert-github** [comment]
@doxygen 
Regarding the `\emoji` command and also the `\ref`, `\subpage` and `\link` commands in LaTeX there are some problems. 
I found a solution already and working on a patch.

**Pull request merged**